utils/serialization/maps/map.go

```diff
@@ -148,11 +151,57 @@ func toTime(f reflect.Type, t reflect.Type, data any) (any, error) {
 	}
 }
 
+func toCustomTypeIntFallback(f reflect.Type, t reflect.Type, data any) (any, error) {
+	if f.Kind() != reflect.String {
+		return data, nil
+	}
+	if t.Kind() != reflect.Int {
+		return data, nil
+	}
+
+	s, ok := data.(string)
+	if !ok {
+		return data, nil
+	}
+
+	i, err := strconv.Atoi(s)
+	if err != nil {
+		return data, nil
+	}
+
+	ptr := reflect.New(t).Elem()
+	ptr.SetInt(safecast.ToInt64(i))
+
+	return ptr.Interface(), nil
+}
+
+func toCustomType(f reflect.Type, t reflect.Type, data any) (any, error) {
+	if f.Kind() != reflect.String {
+		return data, nil
+	}
+
+	customType, ok := reflect.New(t).Interface().(encoding.TextUnmarshaler)
+	if !ok {
+		return toCustomTypeIntFallback(f, t, data)
+	}
+
+	err := customType.UnmarshalText([]byte(data.(string))) // we know it is a string based on reflection
```

Comment:
could we have a way to support optional fields as well as in *string

Reply:
We will do this as another PR where we sort out support for pointers